pkgs/development/lua-modules/generated-packages.nix

```diff
@@ -2977,6 +2977,29 @@ buildLuarocksPackage {
   };
 }) {};
 
+oil-nvim = callPackage({ buildLuarocksPackage, fetchurl, fetchzip, luaOlder, nvim-web-devicons }:
+buildLuarocksPackage {
+  pname = "oil.nvim";
+  version = "2.15.0-1";
+  knownRockspec = (fetchurl {
+    url    = "mirror://luarocks/oil.nvim-2.15.0-1.rockspec";
+    sha256 = "0xkych23rn6jpj4hbam1j7ca1gwb9z3lzfm7id3dvcqj8aysv77j";
+  }).outPath;
+  src = fetchzip {
+    url    = "https://github.com/stevearc/oil.nvim/archive/v2.15.0.zip";
+    sha256 = "0rrv7wg0nwfj5fd6byxs4np1p18xxdzyv11ba6vqqh3s6z0qwawc";
+  };
+
+  disabled = luaOlder "5.1";
+  propagatedBuildInputs = [ nvim-web-devicons ];
```

Comment:
It does _not_ depend on `nvim-web-devicons`, it's an optional dependency (in fact it also supports `mini.icons` as an alternative).

Basically the same issue as https://github.com/nvim-neorocks/nurr/pull/45.

Reply:
See https://github.com/nvim-neorocks/nurr/pull/50.

Reply:
This code is auto generated, it will be updated in the next vimPlugins bump

Reply:
That's because there weren't any releases so far, see https://github.com/stevearc/oil.nvim/releases

I can craft an overlay for installing latest dev version, if you want to

Reply:
Figured, I had noticed that the LuaRocks package versions are suffixed anyway, I kind of expected NURR to update with a `-2` package which dropped the dependency.

Reply:
This is something that should be handled in NURR, not nixpkgs. Please open an issue there

Reply:
Yeah, I agree, I was expecting it to _be_ handled given your original comment :-).